ports/cxd56/common-hal/camera/Camera.c

```diff
@@ -0,0 +1,204 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright 2020 Sony Semiconductor Solutions Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <string.h>
+#include <fcntl.h>
+
+#include <arch/board/board.h>
+#include <nuttx/video/video.h>
+#include <nuttx/arch.h>
+
+#include "py/runtime.h"
+
+#include "shared-bindings/camera/Camera.h"
+
+typedef struct {
+    const char* devpath;
+    int fd;
+} camera_dev_t;
+
+STATIC camera_dev_t camera_dev = {"/dev/video", -1};
+
+static bool camera_check_width_and_height(uint16_t width, uint16_t height) {
+    if ((width == VIDEO_HSIZE_QVGA && height == VIDEO_VSIZE_QVGA) ||
+        (width == VIDEO_HSIZE_VGA && height == VIDEO_VSIZE_VGA) ||
+        (width == VIDEO_HSIZE_HD && height == VIDEO_VSIZE_HD) ||
+        (width == VIDEO_HSIZE_QUADVGA && height == VIDEO_VSIZE_QUADVGA) ||
+        (width == VIDEO_HSIZE_FULLHD && height == VIDEO_VSIZE_FULLHD) ||
+        (width == VIDEO_HSIZE_3M && height == VIDEO_VSIZE_3M) ||
+        (width == VIDEO_HSIZE_5M && height == VIDEO_VSIZE_5M)) {
```

Comment:
Not a big deal, but this could be a table of HSIZEs and VSIZEs that you iterate through to check.